contracts/contracts/strategies/ThreePoolStrategy.sol

```diff
@@ -163,31 +203,23 @@ contract ThreePoolStrategy is InitializableAbstractStrategy {
      */
     function withdrawAll() external onlyVaultOrGovernor nonReentrant {
         // Withdraw all from Gauge
-        (, uint256 gaugePTokens, ) = _getTotalPTokens();
+        (, uint256 gaugePTokens, uint256 totalPTokens) = _getTotalPTokens();
         ICurveGauge(crvGaugeAddress).withdraw(gaugePTokens);
-        // Remove entire balance, 3pool strategies only support a single asset
-        // so safe to use assetsMapped[0]
-        IERC20 asset = IERC20(assetsMapped[0]);
-        uint256 pTokenBalance = IERC20(assetToPToken[address(asset)]).balanceOf(
-            address(this)
-        );
-        uint256 minWithdrawAmount = pTokenBalance.mulTruncate(
+        uint256 minWithdrawAmount = totalPTokens.mulTruncate(
             uint256(1e18).sub(maxSlippage)
         );
-        ICurvePool(platformAddress).remove_liquidity_one_coin(
-            pTokenBalance,
-            poolCoinIndex,
-            minWithdrawAmount
-        );
-        // Transfer the asset out to Vault
-        asset.safeTransfer(vaultAddress, asset.balanceOf(address(this)));
+        ICurvePool threePool = ICurvePool(platformAddress);
+        threePool.remove_liquidity(totalPTokens, minWithdrawAmount);
```

Comment:
@tomlinton, I think `remove_liquidity` takes an array as its second argument. The function definition is:

`def remove_liquidity(_amount: uint256, min_amounts: uint256[N_COINS]):`

..see <https://github.com/curvefi/curve-contract/blob/master/contracts/pools/3pool/StableSwap3Pool.vy#L498>